### PR TITLE
Egypt local flavor

### PR DESCRIPTION
**Pull request by AhShXyZ**
**All Changes**

- [x] Add an entry to the docs/changelog.rst describing the change.

- [x] Add an entry for your name in the docs/authors.rst file if it's not
      already there.

**New Fields Only**

- [x] Prefix the country code to all fields.

- [x] Field names should be easily understood by developers from the target
      localflavor country. This means that English translations are usually
      not the best name unless it's for something standard like postal code,
      tax / VAT ID etc.

- [x] Add meaningful tests. 100% test coverage is not required but all
      validation edge cases should be covered.

- [x] Add `.. versionadded:: <next-version>` comment markers to new
      localflavors.

- [x] Add documentation for all fields.

### CODE REVIEW

**Comment by benkonrath** [comment]
Thanks for the PR! I'll have time to review this in the next few days. Thanks for your patience.

File: localflavor/eg/forms.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            date(int(year), int(month), int(day))
+        except ValueError:
+            raise ValidationError(self.error_messages['invalid'])
+
+        # is valid governorate code?
+        governorate_codes = (
+            '01', '02', '03', '04', '11', '12', '13', '14', '15', '16', '17', '18', '19', '21',
+            '22', '23', '24', '25', '26', '27', '28', '29', '31', '32', '33', '34', '35', '88')
+        if governorate_code not in governorate_codes:
+            raise ValidationError(self.error_messages['invalid'])
```
Comment (benkonrath) [inline localflavor/eg/forms.py]: This invalid case isn't covered by the tests. Can you add an invalid id number to the tests that has an invalid governote code? - Just for completeness. Thanks.

Comment (AhShXyZ) [reply localflavor/eg/forms.py]: Reached 98% code coverage instead of 93% hope that's the desired change. Thank you for your time.

**Comment by benkonrath** [comment]
Thanks again for your contribution!
